### DISCUSSION TITLE
registerNode 如何动态控制是否显示相关元素？

### DISCUSSION THREAD

**Discussion by world56** [antvis/X6#2476] [Q&A] [answered]
<img width="647" alt="image" src="https://user-images.githubusercontent.com/37354312/183548170-c1ea85ae-d007-4c67-993b-52ecc6c4fc94.png">

**Comment by world56** [comment]
有没有类似回调函数的方法？每次render前获取参数进行判断是否显示？

**Reply by zmcode** [reply]
每个节点的数据都会存放到自身的store中,你可以先创建这个节点, 不渲染, 然后根据对应的业务逻辑改变store中的数据渲染即可

**Reply by world56** [reply]
感谢 请问直接更新状态，如何立即绘制显示呢？

target.store.data.ports.items[0].attrs.image.visibility = 'visible'

**Comment by world56** [comment, marked as answer]
找到修改的方法了，找了Issues、Discussions、官方文档半天没找到，看原型链试出来了。🤡

例：
const { height, width } = node.store.get("size");
node.store.set("size", { height: height - 36, width });
